prow/crier/reporters/pubsub/reporter.go

```diff
@@ -130,13 +130,36 @@ func (c *Client) generateMessageFromPJ(pj *prowapi.ProwJob) *ReportMessage {
 	}
 	refs = append(refs, pj.Spec.ExtraRefs...)
 
+	var storagePath string
+	// calculate storagePath if pj.Status.URL is set
+	if pj.Status.URL != "" {
+		// example:
+		// * pj.Status.URL: https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-benchmark-microbenchmarks/1258197944759226371
+		// * prefix: https://prow.k8s.io/view/
+		// * storageURLPath: gs/kubernetes-jenkins/logs/ci-benchmark-microbenchmarks/1258197944759226371
+		prefix := c.config().Plank.GetJobURLPrefix(pj.Spec.Refs)
+
+		storageURLPath := strings.TrimPrefix(pj.Status.URL, prefix)
+		if strings.HasPrefix(storageURLPath, api.GCSKeyType) {
+			storageURLPath = strings.Replace(storageURLPath, api.GCSKeyType, providers.GS, 1)
+		}
+
+		if providers.HasStorageProviderPrefix(storageURLPath) {
+			storagePathSegments := strings.SplitN(storageURLPath, "/", 2)
```

Comment:
So if the path is just "gs" or "s3" we will panic? Can we add a lencheck here? Shouldn't happen ofc but shouldn't kill the whole controller if it does either

Reply:
Done